Cannot compile test suite

Compiling the target `run_tests` fails with both gcc and clang on the file `src/test/score.cpp`.
`cmake -DDEV=ON -DDOWNLOAD_GTEST=ON -DTEST_MYSQL=OFF -GNinja .. && ninja run_tests`

I do have the libgtest installed:
```bash
$ ll /usr/lib/libg*
-rw-r--r-- 1 root root 1,8M mars  21 03:34 libgtest.a
-rw-r--r-- 1 root root 3,9K mars  21 03:34 libgtest_main.a
```

Clang output:
```shell
FAILED: CMakeFiles/testrunner.dir/src/test/score.cpp.o
/usr/local/bin/clang++  -DCONF_INFORM_UPDATE -DCONF_OPENSSL -DCONF_VIDEORECORDER -DGAME_RELEASE_VERSION=\"15.9.1\" -DGLEW_STATIC -D_FORTIFY_SOURCE=2 -isystem /usr/include/x86_64-linux-gnu -Isrc -I../src -O3 -DNDEBUG   -fdiagnostics-color=always -fcolor-diagnostics -fstack-protector-all -fno-exceptions -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wformat=2 -Wno-nullability-completeness -Wthread-safety -std=gnu++1z -MD -MT CMakeFiles/testrunner.dir/src/test/score.cpp.o -MF CMakeFiles/testrunner.dir/src/test/score.cpp.o.d -o CMakeFiles/testrunner.dir/src/test/score.cpp.o -c ../src/test/score.cpp
../src/test/score.cpp:504:1: error: unknown type name 'Sql'
INSTANTIATE(SingleScore);
^
../src/test/score.cpp:494:27: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                                 ^
../src/test/score.cpp:504:1: error: parameter type 'SingleScore' is an abstract class
../src/test/score.cpp:494:37: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                                           ^
/usr/include/gtest/gtest.h:446:16: note: unimplemented pure virtual method 'TestBody' in 'SingleScore'
  virtual void TestBody() = 0;
               ^
../src/test/score.cpp:504:1: error: unknown type name 'g_TestValues'
INSTANTIATE(SingleScore);
^
../src/test/score.cpp:494:39: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                                             ^
../src/test/score.cpp:504:1: error: C++ requires a type specifier for all declarations
../src/test/score.cpp:495:3: note: expanded from macro 'INSTANTIATE'
                [](const testing::TestParamInfo<Score::ParamType> &Info) { \
                ^
../src/test/score.cpp:504:1: error: 'type name' declared as array of functions of type 'int (const testing::TestParamInfo<Score::ParamType> &)' (aka 'int (const TestParamInfo<IDbConnection *> &)')
../src/test/score.cpp:495:3: note: expanded from macro 'INSTANTIATE'
                [](const testing::TestParamInfo<Score::ParamType> &Info) { \
                ^
../src/test/score.cpp:504:1: error: expected ')'
../src/test/score.cpp:495:60: note: expanded from macro 'INSTANTIATE'
                [](const testing::TestParamInfo<Score::ParamType> &Info) { \
                                                                         ^
../src/test/score.cpp:504:1: note: to match this '('
../src/test/score.cpp:494:26: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                                ^
../src/test/score.cpp:504:1: error: C++ requires a type specifier for all declarations
INSTANTIATE(SingleScore);
^
../src/test/score.cpp:494:2: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
        ^
../src/test/score.cpp:505:1: error: unknown type name 'Sql'
INSTANTIATE(TeamScore);
^
../src/test/score.cpp:494:27: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                                 ^
../src/test/score.cpp:505:1: error: parameter type 'TeamScore' is an abstract class
../src/test/score.cpp:494:37: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                                           ^
/usr/include/gtest/gtest.h:446:16: note: unimplemented pure virtual method 'TestBody' in 'TeamScore'
  virtual void TestBody() = 0;
               ^
../src/test/score.cpp:505:1: error: unknown type name 'g_TestValues'
INSTANTIATE(TeamScore);
^
../src/test/score.cpp:494:39: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                                             ^
../src/test/score.cpp:505:1: error: C++ requires a type specifier for all declarations
../src/test/score.cpp:495:3: note: expanded from macro 'INSTANTIATE'
                [](const testing::TestParamInfo<Score::ParamType> &Info) { \
                ^
../src/test/score.cpp:505:1: error: 'type name' declared as array of functions of type 'int (const testing::TestParamInfo<Score::ParamType> &)' (aka 'int (const TestParamInfo<IDbConnection *> &)')
../src/test/score.cpp:495:3: note: expanded from macro 'INSTANTIATE'
                [](const testing::TestParamInfo<Score::ParamType> &Info) { \
                ^
../src/test/score.cpp:505:1: error: expected ')'
../src/test/score.cpp:495:60: note: expanded from macro 'INSTANTIATE'
                [](const testing::TestParamInfo<Score::ParamType> &Info) { \
                                                                         ^
../src/test/score.cpp:505:1: note: to match this '('
../src/test/score.cpp:494:26: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                                ^
../src/test/score.cpp:505:1: error: C++ requires a type specifier for all declarations
INSTANTIATE(TeamScore);
^
../src/test/score.cpp:494:2: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
        ^
../src/test/score.cpp:506:1: error: unknown type name 'Sql'
INSTANTIATE(MapInfo);
^
../src/test/score.cpp:494:27: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                                 ^
../src/test/score.cpp:506:1: error: parameter type 'MapInfo' is an abstract class
../src/test/score.cpp:494:37: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                                           ^
/usr/include/gtest/gtest.h:446:16: note: unimplemented pure virtual method 'TestBody' in 'MapInfo'
  virtual void TestBody() = 0;
               ^
../src/test/score.cpp:506:1: error: unknown type name 'g_TestValues'
INSTANTIATE(MapInfo);
^
../src/test/score.cpp:494:39: note: expanded from macro 'INSTANTIATE'
        INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                                             ^
../src/test/score.cpp:506:1: error: C++ requires a type specifier for all declarations
../src/test/score.cpp:495:3: note: expanded from macro 'INSTANTIATE'
                [](const testing::TestParamInfo<Score::ParamType> &Info) { \
                ^
../src/test/score.cpp:506:1: error: 'type name' declared as array of functions of type 'int (const testing::TestParamInfo<Score::ParamType> &)' (aka 'int (const TestParamInfo<IDbConnection *> &)')
../src/test/score.cpp:495:3: note: expanded from macro 'INSTANTIATE'
                [](const testing::TestParamInfo<Score::ParamType> &Info) { \
                ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

gcc output:
```shell
FAILED: CMakeFiles/testrunner.dir/src/test/score.cpp.o
/usr/bin/g++  -DCONF_INFORM_UPDATE -DCONF_OPENSSL -DCONF_VIDEORECORDER -DGAME_RELEASE_VERSION=\"15.9.1\" -DGLEW_STATIC -D_FORTIFY_SOURCE=2 -isystem /usr/include/x86_64-linux-gnu -Isrc -I../src -O3 -DNDEBUG   -fdiagnostics-color=always -fstack-protector-all -fno-exceptions -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wformat=2 -Wno-nullability-completeness -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wshadow=local -std=gnu++1z -MD -MT CMakeFiles/testrunner.dir/src/test/score.cpp.o -MF CMakeFiles/testrunner.dir/src/test/score.cpp.o.d -o CMakeFiles/testrunner.dir/src/test/score.cpp.o -c ../src/test/score.cpp
../src/test/score.cpp:494:26: error: expected constructor, destructor, or type conversion before ‘(’ token
  INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                          ^
../src/test/score.cpp:504:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(SingleScore);
 ^~~~~~~~~~~
../src/test/score.cpp:502:4: error: expected unqualified-id before ‘)’ token
   })
    ^
../src/test/score.cpp:504:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(SingleScore);
 ^~~~~~~~~~~
../src/test/score.cpp:494:26: error: expected constructor, destructor, or type conversion before ‘(’ token
  INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                          ^
../src/test/score.cpp:505:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(TeamScore);
 ^~~~~~~~~~~
../src/test/score.cpp:502:4: error: expected unqualified-id before ‘)’ token
   })
    ^
../src/test/score.cpp:505:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(TeamScore);
 ^~~~~~~~~~~
../src/test/score.cpp:494:26: error: expected constructor, destructor, or type conversion before ‘(’ token
  INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                          ^
../src/test/score.cpp:506:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(MapInfo);
 ^~~~~~~~~~~
../src/test/score.cpp:502:4: error: expected unqualified-id before ‘)’ token
   })
    ^
../src/test/score.cpp:506:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(MapInfo);
 ^~~~~~~~~~~
../src/test/score.cpp:494:26: error: expected constructor, destructor, or type conversion before ‘(’ token
  INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                          ^
../src/test/score.cpp:507:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(MapVote);
 ^~~~~~~~~~~
../src/test/score.cpp:502:4: error: expected unqualified-id before ‘)’ token
   })
    ^
../src/test/score.cpp:507:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(MapVote);
 ^~~~~~~~~~~
../src/test/score.cpp:494:26: error: expected constructor, destructor, or type conversion before ‘(’ token
  INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                          ^
../src/test/score.cpp:508:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(Points);
 ^~~~~~~~~~~
../src/test/score.cpp:502:4: error: expected unqualified-id before ‘)’ token
   })
    ^
../src/test/score.cpp:508:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(Points);
 ^~~~~~~~~~~
../src/test/score.cpp:494:26: error: expected constructor, destructor, or type conversion before ‘(’ token
  INSTANTIATE_TEST_SUITE_P(Sql, SUITE, g_TestValues, \
                          ^
../src/test/score.cpp:509:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(RandomMap);
 ^~~~~~~~~~~
../src/test/score.cpp:502:4: error: expected unqualified-id before ‘)’ token
   })
    ^
../src/test/score.cpp:509:1: note: in expansion of macro ‘INSTANTIATE’
 INSTANTIATE(RandomMap);
 ^~~~~~~~~~~
cc1plus: warning: unrecognized command line option ‘-Wno-nullability-completeness’
```

I got this:  ../src/test/score.cpp:1:10: fatal error: gmock/gmock.h: No such file or directory
installing gmock-devel  [1] solved the problem for me 

[1] 
```
dnf install /usr/include/gmock/gmock.h
Installing:
gmock-devel                            x86_64                            1.10.0-6.fc34                             updates                            145 k
Installing dependencies:
gmock                                  x86_64                            1.10.0-6.fc34                             updates                             59 k
gtest-devel                            x86_64                            1.10.0-6.fc34                             updates                            170 k

--- a/README.md
+++ b/README.md
@@ -36,6 +36,8 @@ You can install the required libraries on your system, `touch CMakeLists.txt` an
 
     sudo apt install build-essential cmake git google-mock libcurl4-openssl-dev libssl-dev libfreetype6-dev libglew-dev libnotify-dev libogg-dev libopus-dev libopusfile-dev libpnglite-dev libsdl2-dev libsqlite3-dev libwavpack-dev python libx264-dev libavfilter-dev libavdevice-dev libavformat-dev libavcodec-extra libavutil-dev libvulkan-dev glslang-tools spirv-tools
 
+On older distributions like Ubuntu 18.04 don't install `google-mock`, but instead set `-DDOWNLOAD_GTEST=ON` when building to get a more recent gtest/gmock version.
+
 Or on CentOS, RedHat and AlmaLinux like this:
 
     sudo yum install gcc gcc-c++ make cmake git python2 gtest-devel gmock-devel libcurl-devel openssl-devel freetype-devel glew-devel libnotify-devel libogg-devel opus-devel opusfile-devel pnglite-devel SDL2-devel sqlite-devel wavpack-devel libx264-devel ffmpeg-devel vulkan-devel glslang spirv-tools

